### DISCUSSION TITLE
Unused 'php'

### DISCUSSION THREAD

**Discussion by MGatner** [composer-unused/composer-unused#227] [Q&A] [answered]
`composer-unused` is failing in my CI due to `php` being an unused package. I assume this is because my **composer.json** includes the PHP version in the `require` (`"php": "^7.3 || ^8.0"`), but that has to be a common practice so something else must be wrong here? Here's an example run: https://github.com/tattersoftware/codeigniter4-roster/runs/3997892004?check_suite_focus=true

**Comment by icanhazstring** [comment, marked as answer]
Hi, just answered it here :)
https://github.com/composer-unused/composer-unused/discussions/226#discussioncomment-1532563

This is a known issue that will be fixed in the next release.

**Reply by MGatner** [reply]
Ah thanks! GitHub didn't notify me of your answer. Sorry for the duplicate - it was very hard to search for solutions on "php" :)

**Comment by MGatner** [comment]
For now I'm able to work around this by including `--excludePackage=php`.

**Comment by MGatner** [comment]
This issue has cropped back up for me using the `main` branch **bin/composer-unused**:
```
Results
-------

Found 0 used, 1 unused, 15 ignored and 0 zombie packages

 Used packages

 Unused packages
 ✗ php

```

**Reply by MGatner** [reply]
@icanhazstring

**Reply by icanhazstring** [reply]
This is kind of "hacky" fix 😉
Which I need to change as well. You need to have at least one "real" dependency. If you ignore everything php will be unused again.

Will probably change that in the future 👌

**Reply by MGatner** [reply]
Gotcha! No worries then - I wasn't done defining these dependency files, and it will have some. Glad this wasn't a regression 💪
